packages/kbn-reporting/public/share/share_context_menu/register_pdf_png_modal_reporting.tsx

```diff
@@ -334,11 +335,73 @@ export const reportingExportModalProvider = ({
         });
     };
 
-    const generateReportPNG = ({ intl }: { intl: InjectedIntl }) => {
-      const el = document.querySelector('[data-shared-items-container]');
-      const { height, width } = el ? el.getBoundingClientRect() : { height: 768, width: 1024 };
-      const dimensions = { height, width };
+    const generateReportPDFForPrinting = ({ intl }: { intl: InjectedIntl }) => {
+      const { layout: outerLayout } = getJobParams(jobProviderOptions, 'printablePdfV2')();
+      let dimensions = outerLayout?.dimensions;
+      if (!dimensions) {
+        const el = document.querySelector('[data-shared-items-container]');
+        const { height, width } = el ? el.getBoundingClientRect() : { height: 768, width: 1024 };
+        dimensions = { height, width };
+      }
+      const decoratedJobParams = apiClient.getDecoratedJobParams({
+        ...getJobParams(jobProviderOptions, 'printablePdfV2')(),
+        layout: { id: 'print', dimensions },
+        objectType,
+        title: sharingData.title,
+      });
+      return apiClient
+        .createReportingJob('printablePdfV2', decoratedJobParams)
+        .then(() => {
+          toasts.addSuccess({
+            title: intl.formatMessage(
+              {
+                id: 'reporting.share.modalContent.successfullyQueuedReportNotificationTitle',
+                defaultMessage: 'Queued report for {objectType}',
+              },
+              { objectType }
+            ),
+            text: toMountPoint(
+              <FormattedMessage
+                id="reporting.share.modalContent.successfullyQueuedReportNotificationDescription"
+                defaultMessage="Track its progress in {path}."
+                values={{
+                  path: (
+                    <a href={apiClient.getManagementLink()}>
+                      <FormattedMessage
+                        id="reporting.share.publicNotifier.reportLink.reportingSectionUrlLinkLabel"
+                        defaultMessage="Stack Management &gt; Reporting"
+                      />
+                    </a>
+                  ),
+                }}
+              />,
+              { theme, i18n: i18nStart }
+            ),
+            'data-test-subj': 'queueReportSuccess',
+          });
+          if (onClose) {
+            onClose();
+          }
+        })
+        .catch((error: any) => {
+          toasts.addError(error, {
+            title: intl.formatMessage({
+              id: 'reporting.share.modalContent.notification.reportingErrorTitle',
+              defaultMessage: 'Unable to create report',
+            }),
+            toastMessage: error.body?.message,
+          });
+        });
+    };
```

Comment:
Do we have a reason for including this, the function body for `generateReportPDFForPrinting` is a duplicate of the `generateReportPDF` function with only a differentiation in the the layout param?

Reply:
I had to change this back because it didn't work when the radio buttons were changed when the modal was rendered.